### PR TITLE
Fix DependencyModel tests on NativeAOT

### PR DESCRIPTION
**Pull request by eerhardt**

### CODE REVIEW

File: src/libraries/Microsoft.Extensions.DependencyModel/tests/DependencyContextTests.cs [hunk 1]
```diff
@@ -280,13 +281,20 @@ public void MergeMergesRuntimeGraph()
...
         public void DefaultWorksCorrectly()
         {
-            // only need to assert the context contains non-null properties.
-
             var context = DependencyContext.Default;
-            Assert.NotNull(context);
-            Assert.NotNull(context.RuntimeGraph);
-            Assert.NotNull(context.RuntimeLibraries);
-            Assert.NotNull(context.Target);
+            if (PlatformDetection.IsNativeAot)
```
Comment (MichalStrehovsky) [inline src/libraries/Microsoft.Extensions.DependencyModel/tests/DependencyContextTests.cs:285]: Does this work for single file deployments in general? If not:

```suggestion
            if (PlatformDetection.IsSingleFile)
```

**Review by MichalStrehovsky — APPROVED** [review]

File: src/libraries/Microsoft.Extensions.DependencyModel/tests/DependencyContextTests.cs (hunk 1 above)
Comment (eerhardt) [reply src/libraries/Microsoft.Extensions.DependencyModel/tests/DependencyContextTests.cs]: Good call. Fixed.

**Review by akoeplinger — APPROVED** [review]

**Comment by akoeplinger** [comment]
Failures are unrelated.

**Pull request merged**